packages/session/src/sessions.ts

```diff
@@ -62,6 +65,29 @@ export class Sessions implements ISessions {
       path: path ?? session.path,
       name: name ?? session.name,
     };
+
+    if (kernel) {
+      // Kernel id takes precedence over name.
+      if (kernel?.id) {
+        const session = this._sessions.find(
+          (session) => session.kernel?.id === kernel?.id
+        );
+        if (session) {
+          patched['kernel'] = session?.kernel;
```

Comment:
Wondering if it's possible to use the following notation?

```suggestion
          patched.kernel = session.kernel;
```

Normally TypeScript should be able to infer the type based on the function return type.